src/librustc_mir/borrow_check/nll/region_infer/error_reporting/mod.rs

```diff
@@ -54,6 +54,31 @@ enum Trace {
     NotVisited,
 }
 
+/// Various pieces of state used when reporting borrow checker errors.
+pub struct ErrorReportingCtx<'a, 'b, 'tcx> {
+    #[allow(dead_code)] // FIXME(mark-i-m): used by outlives suggestions
+    rinfcx: &'b RegionInferenceContext<'tcx>,
+    infcx: &'b InferCtxt<'a, 'tcx>,
+
+    mir_def_id: DefId,
+    body: &'b Body<'tcx>,
+    upvars: &'b [Upvar],
+}
+
+/// Information about the various region constraints involved in a borrow checker error.
+#[derive(Clone, Debug)]
+pub struct ErrorConstraintInfo {
```

Comment:
Would be good to document the other fields also?

Reply:
:+1: will do